drawBot/context/baseContext.py

```diff
@@ -537,7 +537,34 @@ def transform(self, transformMatrix, center=(0, 0)):
         aT.setTransformStruct_(transformMatrix[:])
         self._path.transformUsingAffineTransform_(aT)
 
-    # boolean operations
+    def fit(self, bounds):
```

Comment:
I propose to extend this:
```python
    def fit(self, bounds, constrain=True):
        ...
```
Where `constrain=True` behaves like what you have implemented, and `False` would mean: fill up the whole bounds.

Maybe `bounds` should be called `box`, and have the same meaning as in `textBox()`: `(x, y, w, h)`, as this is more common in DrawBot overall.

Reply:
Or perhaps `constrainProportions`.

Reply:
I agree with box instead of bounds. I used bounds because it can be used to align one shape to another, but it could also be confusing.

Could you please elaborate on fill up bounds, I didn't get the idea.

Reply:
You now scale by a factor that will make the shape fit in the box, while maintaining widht/height proportions (constrain). You could also scale x and y independently, so the bounding box of the shape will fill the entire target box. That's what I mean with `constrainProportions=False`. Hope that clears it up.

Reply:
Ow I see! Like keeping the ratio.